### PR TITLE
Fix #128 Additional compiler option to control the groovy compiler

### LINKED ISSUE #128: Control the groovyc compiler
It would be nice to be able to pass options to the `groovyc` ant task in `InstrumentCodeAction` 

Pull request coming up. :slightly_smiling_face:

### PR DESCRIPTION
**Pull request by sbglasius**
Add option to pass arguments to the `groovyc` compiler

### CODE REVIEW

File: src/functTest/projects/groovy-project-compile-config/build.gradle [hunk 1]
```diff
@@ -27,6 +27,7 @@ tasks.withType(GroovyCompile) {
 clover {
     compiler {
         encoding = 'UTF-8'
+        additionalGroovycOpts = [listfiles: 'yes']
```
Comment (Alex-Vol-SV) [inline src/functTest/projects/groovy-project-compile-config/build.gradle]: Fantastic. This is a good change. Perhaps adding an assertion in the unit test to validate it has done the work?

File: README.md [hunk 2]
```diff
@@ -163,6 +163,7 @@ need to be carried over to the compilation of the instrumented sources.  These a
 * `executable`: The (optional) javac executable to use, should be an absolute file.
 * `debug`: Controls whether to invoke javac with the -g flag. This is useful for Spring MVC code that uses reflection for parameter mapping. (defaults to `false`).
 * `additionalArgs`: The (optional) additional command line arguments for the compiler. This is useful for Spring MVC code that uses reflection for parameter mapping. This should be valid command line arguments as a spaces separated string. No attempt is made to validate this line, it is passed verbatim to the <compilerarg> nested element for the Ant `javac` task.
+* `additionalGroovycOpts`: The (optional) additional options for the `groovyc` compiler. See [http://groovy-lang.org/groovyc.html#_ant_task]
```
Comment (Alex-Vol-SV) [inline README.md]: Thank you for taking care of the docs

File: src/main/groovy/com/bmuschko/gradle/clover/InstrumentCodeAction.groovy [hunk 3]
```diff
@@ -310,7 +310,8 @@ class InstrumentCodeAction implements Action<Task> {
     private void compileGroovyAndJava(AntBuilder ant, Collection<File> srcDirs, File destDir, String classpath) {
         if (srcDirs.size() > 0) {
             String args = getAdditionalArgs()
-            ant.groovyc(destdir: destDir.canonicalPath, classpath: classpath, encoding: getEncoding()) {
+            Map groovycAttrs = [destdir: destDir.canonicalPath, classpath: classpath, encoding: getEncoding()] + (getAdditionalGroovycOpts() ?: [:])
+            ant.groovyc(groovycAttrs) {
```
Comment (Alex-Vol-SV) [inline src/main/groovy/com/bmuschko/gradle/clover/InstrumentCodeAction.groovy]: Simple, elegant. I will pull this in the 2.2.2 patch by the end of the week. Could you see if you can make a validation assertion that the added arguments did something in the functional test? Perhaps examining the task output?

Comment (sbglasius) [reply src/main/groovy/com/bmuschko/gradle/clover/InstrumentCodeAction.groovy]: Yes, I'll see what I can do.

**Comment by sbglasius** [comment]
@Alex-Vol-SV Fixed it, but had to make the `gradle` output more verbose (`--info` flag)

File: src/functTest/projects/groovy-project-compile-config/build.gradle (hunk 1 above)
Comment (sbglasius) [reply src/functTest/projects/groovy-project-compile-config/build.gradle]: Done.

File: src/functTest/groovy/com/bmuschko/gradle/clover/GroovyProjectCompileConfigSpec.groovy [hunk 4]
```diff
@@ -47,6 +48,12 @@ class GroovyProjectCompileConfigSpec extends AbstractFunctionalTestBase {
         and: "the Clover snapshot is not generated because test optimization is not enabled"
         cloverSnapshot.exists() == false
 
+        and: "that the gradle output contains the two files compiled by groovyc"
+        with(result.output) {
+            contains('Book.groovy')
+            contains('BookGroovyTest.groovy')
+        }
+
```
Comment (Alex-Vol-SV) [inline src/functTest/groovy/com/bmuschko/gradle/clover/GroovyProjectCompileConfigSpec.groovy]: Very nice

**Pull request merged**